File: lcof/面试题57. 和为s的两个数字/Solution.go
func twoSum(nums []int, target int) []int {
	l, r := 0, len(nums)-1
	for {
		if nums[l]+nums[r] == target {
			return []int{nums[l], nums[r]}
		}
		if nums[l]+nums[r] > target {
			r--
		} else {
			l++
		}
	}
}

File: lcof/面试题57. 和为s的两个数字/Solution.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int l = 0, r = nums.Length - 1;
        while (true) {
            if (nums[l] + nums[r] == target) {
                return new int[] {nums[l], nums[r]};
            }
            if (nums[l] + nums[r] > target) {
                --r;
            } else {
                ++l;
            }
        }
    }
}
